libs/types/assisted-installer-service.d.ts

```diff
@@ -2189,13 +2190,27 @@ export type OsImages = OsImage[];
  */
 export interface Platform {
   type: PlatformType;
+  external?: PlatformExternal;
   /**
    * Used by the service to indicate that the platform-specific components are not included in
    * OpenShift and must be provided as manifests separately.
    */
   readonly isExternal?: boolean;
 }
-export type PlatformType = 'baremetal' | 'nutanix' | 'vsphere' | 'none' | 'oci';
+/**
+ * Configuration used when installing with an external platform type.
+ */
+export interface PlatformExternal {
+  /**
+   * Holds the arbitrary string representing the infrastructure provider name.
+   */
+  platformName?: string;
+  /**
+   * When set to external, this property will enable an external cloud provider.
+   */
+  cloudControllerManager?: '' | 'External';
+}
+export type PlatformType = 'baremetal' | 'nutanix' | 'vsphere' | 'none' | 'oci' | 'external';
```

Comment:
The ticket indicates that for example for OCI, we should be taking it from the "external" object.
But this MR only seems to add the new types, not to use them.

Is this expected?
Is the data for OCI still returned by the API with `platformType = oci` or would it be migrated to `external` with additional parameters?

Reply:
@celdrake  Yes, it's expected. Bu we are waiting to this[ BE PR ](https://github.com/openshift/assisted-service/pull/5787)to be merged.